src/main/java/be/syntra/devshop/DevshopFront/services/SearchServiceImpl.java

```diff
@@ -81,11 +87,44 @@ private boolean hasSearchRequest() {
     }
 
     @Override
-    public void setPriceFilters(BigDecimal minPrice,BigDecimal maxPrice) {
+    public void setPriceFilters(BigDecimal minPrice, BigDecimal maxPrice) {
         searchModel.setPriceLow(minPrice);
         searchModel.setPriceHigh(maxPrice);
     }
 
+    @Override
+    public void setSelectedCategory(String category) {
+        List<String> selectedCategories = getSelectedCategoriesList();
+        selectedCategories.add(category);
+        searchModel.setSelectedCategories(selectedCategories);
+    }
```

Comment:
if you select a category it is added below the dropdown, if you then refresh the page it is added again, AND it is added to the dropdown too, so it will appear twice in the dropdown and twice below it.  Can this be solved by doing a check here that it's not already in the list wehn trying to add it?

Reply:
I'm genuinely delighted that you found that anomaly! (I didn't). Working on convertion to Set if Thymeleaf can handle it, otherwise a check it'll be

Reply:
converted to Set. Also fixed removal-bug (remove always took first element from list)